packages/flutter/lib/src/widgets/scrollable.dart

```diff
@@ -19,6 +19,24 @@ import 'page_storage.dart';
 import 'scroll_behavior.dart';
 import 'scroll_configuration.dart';
 
+/// Identifies one or both limits of a [Scrollable] in terms of its scrollDirection.
+enum ScrollableEdge {
+  /// The top and bottom of the scrollable if its scrollDirection is vertical
+  /// or the left and right if its scrollDirection is horizontal.
+  both,
+
+  /// Only the top of the scrollable if its scrollDirection is vertical,
+  /// or only the left if its scrollDirection is horizontal.
+  leading,
```

Comment:
I don't have a strong opinion here. Generally, in LTR text, start is the left hand side, end is the right hand side, before is above, and after is below. I think it might be confusing to use start/end here because you might be in an RTL context with a reversed vertical scroller in which case "start" would mean the bottom edge and that wouldn't make any sense. So I can see value in leading/trailing.